routing/pathfind.go

```diff
@@ -1150,6 +1150,10 @@ type blindedPathRestrictions struct {
 	// path. This doesn't include our node, so if the maximum is 1, then
 	// the path will contain our node along with an introduction node hop.
 	maxNumHops uint8
+
+	// nodeOmissionSet holds a set of node IDs of nodes that we should
+	// ignore during blinded path selection.
+	nodeOmissionSet map[route.Vertex]bool
```

Comment:
Can make this `fn.Set[route.Vertex]`, basically aliases to the same underlying map structure, but has nicer helper methods.